main/pipeline/pipeline.cc

```diff
@@ -167,6 +167,7 @@ ast::ParsedFile indexOne(const options::Options &opts, core::GlobalState &lgs, c
                          unique_ptr<KeyValueStore> &kvstore) {
     auto &print = opts.print;
     ast::ParsedFile dslsInlined{nullptr, file};
+    ENFORCE(file.data(lgs).strictLevel == decideStrictLevel(lgs, file, opts));
```

Comment:
I don't think that it would make sense to mutate `strictLevel` in `indexOne` because that would require `indexOne` to unfreeze the file table.

If it does, though, I can move the update to `indexOne`.